app/controllers/works_controller.rb

```diff
@@ -0,0 +1,92 @@
+class WorksController < ApplicationController
+
+  def index
+    @albums = Work.sort_works('album')
+    @books = Work.sort_works('book')
+    @movies = Work.sort_works('movie')
+  end
+
+  def show
+    if @work = Work.find_by(id: params[:id])
+      return @work
+    else
```

Comment:
You repeat this code of finding the work and checking the result many times in this controller. This might be a good place to use a filter.